tiledb/sm/rest/curl.cc

```diff
@@ -187,15 +187,16 @@ Curl::Curl()
 
 Status Curl::init(
     const Config* config,
-    const std::unordered_map<std::string, std::string>& extra_headers) {
+    const std::unordered_map<std::string, std::string>& extra_headers,
+    const std::unordered_map<std::string, std::string>& res_headers) {
   if (config == nullptr)
     return LOG_STATUS(
         Status::RestError("Error initializing libcurl; config is null."));
 
   config_ = config;
   curl_.reset(curl_easy_init());
   extra_headers_ = extra_headers;
-
+  res_headers_ = res_headers;
```

Comment:
`res_headers` is a const-ref, so this will always make a copy. When `Curl::res_headers_` is mutated, it will not mutate `Rest::res_headers_`.

I believe what we want to do here is to leave `Rest::res_headers_` as-is but pass it as a pointer here (e.g. `curlc.init(config, extra_headers_, &res_headers_)` from inside the `Rest` implementation). Modify `Curl::res_headers_` to become a pointer.